.github/actions/setup-candid/action.yaml

```
@@ -0,0 +1,49 @@
+name: Set up candid
+description: Set up candid with provided configuration.
+inputs:
+  candid-location:
+    description: Location to run candid, which will be written to candid's configuration.
+    required: false
+    default: http://127.0.0.1:8081
+  controller-name:
+    description: The name of the controller to bootstrap.
+    required: false
+    default: test
+
+runs:
+  using: composite
+  steps:
+    - name: Install candid
+      shell: bash
+      run: |
+        sudo snap install candid
+
+        while [ -n "$(snap changes candid 2>/dev/null | awk '/^[0-9]+/ {if ($2 != "Done") print $2 }')" ]; do
+          echo "Waiting for snap changes on candid to finish"
+          sleep 1
+        done
+
+        sleep 1
+    - name: Configure candid
+      shell: bash
+      run: |
+        CANDID_CONFIG="/var/snap/candid/current/config.yaml"
+        TMP_CANDID_CONFIG="/tmp/snap-candid-config.yaml"
+        sudo cat "$CANDID_CONFIG" | yq '.location = "${{ inputs.candid-location }}"' | sudo tee "$TMP_CANDID_CONFIG"
+        sudo mv "$TMP_CANDID_CONFIG" "$CANDID_CONFIG"
```

Possibly could use `yq` to write directly to the file e.g.

```suggestion
      run: yq -i '.location = "${{ inputs.candid-location }}"' /var/snap/candid/current/config.yaml
```